Fix a memory leak in EC_GROUP_get_ecparameters()

rhenium opened the pull request:
The variable 'buffer', allocated by EC_POINT_point2buf(), isn't
free'd on the success path.

richsalz commented:
+1

ekasper commented:
A memory leak on a _success_ path really should be caught by tests... Would you mind adding a test to ectest?

mattcaswell commented:
Ping @rhenium - are you able to respond to @ekasper's comment? I'd like to get this in for the 1.1.0 release.

rhenium commented:
I am sorry, I've missed your comment. Added to prime_field_tests(), but I'm not sure if I'm on the right course. Feel free to discard if not.

mattcaswell commented on util/libcrypto.num:
```diff
@@ -4193,3 +4193,7 @@ EVP_PKEY_get1_tls_encodedpoint          4139	1_1_0	EXIST::FUNCTION:
+ECPKPARAMETERS_free                     4143	1_1_0	EXIST::FUNCTION:EC
+ECPARAMETERS_free                       4144	1_1_0	EXIST::FUNCTION:EC
+ECPKPARAMETERS_new                      4145	1_1_0	EXIST::FUNCTION:EC
+ECPARAMETERS_new                        4146	1_1_0	EXIST::FUNCTION:EC
```
These new functions need to be documented in the pod files.

richsalz replied:
they are auto-generated ASN1 functions, which we haven't been documenting.

mattcaswell replied:
Really why not? Perhaps we could get away with a generic page for all the ASN1 functions (if so I agree that would be out of scope for this PR)

richsalz replied:
Because we have not made the ASN1 stuff officially part of the public API.

mattcaswell replied:
Huh? Yeah...I think we have!

mattcaswell commented on test/ectest.c:
```diff
@@ -876,6 +876,23 @@ static void prime_field_tests(void)
+            ABORT;
+
+        EC_GROUP_free(group2);
+        ECPARAMETERS_free(ecparameters);
```
This really has nothing to do with prime field testing. I would add this test as a separate function (e.g. `paramater_tests()` or something), and add a call to it after the `internal_curve_test()` call in `main`.

rhenium commented:
Updated. I added these auto-generated alloc functions in doc/crypto/X509_dup.pod as other functions declared by DECLARE_ASN1_ALLOC_FUNCTIONS are already in it.

mattcaswell commented:
+1

mattcaswell commented:
@richsalz - please reconfirm your plus one.

richsalz commented:
+1.  you merging, @mattcaswell ?

mattcaswell commented:
Merged. Thanks.
